cogeo_mosaic/mosaic.py

```diff
@@ -0,0 +1,241 @@
+
+        if not quiet:
+            click.echo(f"Feed Quadkey index", err=True)
+
+        # Create tree and find assets that overlap each tile
+        tree = STRtree(dataset_geoms)
+
+        for tile in tiles:
+            quadkey = mercantile.quadkey(tile)
+            tile_geom = polygons(mercantile.feature(tile)["geometry"]["coordinates"][0])
+
+            # Find intersections from rtree
+            intersections_idx = sorted(tree.query(tile_geom, predicate="intersects"))
+            if len(intersections_idx) == 0:
+                continue
+
+            intersect_dataset, intersect_geoms = zip(
+                *[(features[idx], dataset_geoms[idx]) for idx in intersections_idx]
+            )
+
+            dataset = cls._filter(tile, intersect_dataset, intersect_geoms, **kwargs)
+
+            if dataset:
+                mosaic_definition["tiles"][quadkey] = [accessor(f) for f in dataset]
+
+        return cls(**mosaic_definition)
+
+    @classmethod
+    def from_urls(
+        cls,
+        urls: Sequence[str],
+        minzoom: Optional[int] = None,
+        maxzoom: Optional[int] = None,
+        max_threads: int = 20,
+        quiet: bool = True,
+        **kwargs,
+    ):
+        """
+        Create mosaicjson from url of COGs.
+
+        Attributes
+        ----------
+        urls : List, required
+            List of COG urls.
+        minzoom: int, optional
+            Force mosaic min-zoom.
+        maxzoom: int, optional
+            Force mosaic max-zoom.
+        max_threads : int
+            Max threads to use (default: 20).
+        quiet: bool, optional (default: True)
+            Mask processing steps.
+        kwargs: any
+            Options forwarded to MosaicJSON.from_features
+
+        Returns
+        -------
+        mosaic_definition : MosaicJSON
+            Mosaic definition.
+
+        """
+        features = get_footprints(urls, max_threads=max_threads, quiet=quiet)
+
+        if minzoom is None:
+            minzoom = {feat["properties"]["minzoom"] for feat in features}
+            if len(minzoom) > 1:
+                warnings.warn(
+                    "Multiple MinZoom, Assets different minzoom values", UserWarning
+                )
+
+            minzoom = max(minzoom)
+
+        if maxzoom is None:
+            maxzoom = {feat["properties"]["maxzoom"] for feat in features}
+            if len(maxzoom) > 1:
+                warnings.warn(
+                    "Multiple MaxZoom, Assets have multiple resolution values",
+                    UserWarning,
+                )
+
+            maxzoom = max(maxzoom)
+
+        datatype = {feat["properties"]["datatype"] for feat in features}
+        if len(datatype) > 1:
+            raise Exception("Dataset should have the same data type")
+
+        return cls._create_mosaic(
+            features, minzoom=minzoom, maxzoom=maxzoom, quiet=quiet, **kwargs
+        )
+
+    @classmethod
+    def from_features(cls, features: Sequence[Dict], **kwargs):
```

Comment:
You really can just set everything here as ``**kwargs``. Just include the full list of attributes in the docstring.

Reply:
Nevermind, `features` allows it to be positional.